discretisedfield/plotting/mpl_field.py

```diff
@@ -872,9 +872,87 @@ def __dir__(self):
 
         return dirlist
 
-    def _add_colorbar(self, ax, cp, colorbar_label):
+    def _add_colorbar(
+        self,
+        ax,
+        cp,
+        colorbar_label,
+        min_height_inches=2.0,
+        min_width_inches=0.5,
+        min_pad_inches=0.5,
+    ):
+        """Adds a colorbar to the current plot.
+
+        This colorbar has a minimum width, height, and padding. Multiple colorbars
+        can be added. These additional colorbars will have the same width and height
+        as the original colorbar.
+        """
+
+        # Need to normalise physical units
+        fig = ax.figure
+        fig_width_inches, fig_height_inches = fig.get_size_inches()
+
+        pos = ax.get_position()
+
+        # relative to ax
+        min_height_norm = min_height_inches / (fig_height_inches * (pos.y1 - pos.y0))
+        # relative to fig
+        min_width_norm = min_width_inches / fig_width_inches
+        min_pad_norm = min_pad_inches / fig_width_inches
+
+        # Calculate pad and width
+        if min_pad_norm > 0.05:
+            pad_h = Size.Fixed(min_pad_inches)
+        else:
+            pad_h = Size.AxesX(ax, aspect=0.05)
+
+        if min_width_norm > 0.05:
+            width_h = Size.Fixed(min_width_inches)
+        else:
+            width_h = Size.AxesX(ax, aspect=0.05)
+
+        if min_height_norm > 1:
+            v_aspect = min_height_norm
+        else:
+            v_aspect = 1
+
+        existing_colorbars = [
+            a for a in fig.get_axes() if f"cb_{id(ax)}" in a.get_label()
+        ]
+
         divider = make_axes_locatable(ax)
-        cax = divider.append_axes("right", size="5%", pad=0.05)
+        # divider = SubplotDivider(fig, 2, 2, 2, aspect=True)
+
+        # axes for colorbar
+        # (the label prevents Axes.add_axes from incorrectly believing that the two
+        # axes are the same)
+        cax = fig.add_axes(
+            divider.get_position(), label=f"cb_{id(ax)}_{len(existing_colorbars)}"
+        )
+
+        # Set the divider to consider the current axis,
+        # previous colorbars, padding and the new colorbar width
+        if len(existing_colorbars) == 0:
+            h = [Size.AxesX(ax), pad_h, width_h]
+            divider.set_horizontal(h)
+        else:
+            divider.new_horizontal(pad_h, pack_start=False)
+            divider.new_horizontal(width_h, pack_start=False)
+
+        v = [Size.AxesY(ax, aspect=v_aspect)]
+
+        # divider.set_horizontal(h)
+        divider.set_vertical(v)
+
+        ax.set_axes_locator(divider.new_locator(nx=0, ny=0))
+
+        for i, cb in enumerate(existing_colorbars):
+            cb.set_axes_locator(divider.new_locator(nx=2 * (i + 1), ny=0))
+        cax.set_axes_locator(
+            divider.new_locator(nx=2 * (len(existing_colorbars) + 1), ny=0)
+        )
```

Comment:
What is done in this part of the code?

Reply:
So when we add new axes, we go through and associate the each old axes with a location as a index and finally the new colour bar